Error when clearing pathfinders from passive mobs on 1.17.1

### Bug Type

Compatibility Issues, Console Errors

### MobChip Version

1.9.2-SNAPSHOT

### Server Version

This server is running CraftBukkit version 3284a-Spigot-3892929-0ab8487 (MC: 1.17.1)

### Description

When trying to clear pathfinders from passive mobs, including cows, pigs, and sheep, an error is printed in console.

### Expected Behavior

Pathfinders should be cleared, and no error should occur

### Actual Behavior

This error is printed in console:
```
Caused by: java.lang.NullPointerException: Cannot read the array length because "array" is null
        at java.util.Arrays.stream(Arrays.java:5427) ~[?:?]
        at be.isach.ultracosmetics.shaded.mobchip.abstraction.v1_17_R1.ChipUtil1_17_R1.fromNMS(ChipUtil1_17_R1.java:1170) ~[?:?]
        at be.isach.ultracosmetics.shaded.mobchip.abstraction.v1_17_R1.ChipUtil1_17_R1.fromNMS(ChipUtil1_17_R1.java:1374) ~[?:?]
        at be.isach.ultracosmetics.shaded.mobchip.abstraction.v1_17_R1.ChipUtil1_17_R1.lambda$getGoals$0(ChipUtil1_17_R1.java:136) ~[?:?]
        at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
        at be.isach.ultracosmetics.shaded.mobchip.abstraction.v1_17_R1.ChipUtil1_17_R1.getGoals(ChipUtil1_17_R1.java:136) ~[?:?]
        at be.isach.ultracosmetics.shaded.mobchip.bukkit.BukkitAI.updateMap(BukkitAI.java:36) ~[?:?]
        at be.isach.ultracosmetics.shaded.mobchip.bukkit.BukkitAI.<init>(BukkitAI.java:31) ~[?:?]
        at be.isach.ultracosmetics.shaded.mobchip.bukkit.BukkitBrain.getGoalAI(BukkitBrain.java:115) ~[?:?]
        at be.isach.ultracosmetics.cosmetics.pets.Pet.clearPathfinders(Pet.java:128) ~[?:?]
        at be.isach.ultracosmetics.cosmetics.pets.Pet.initializeEntity(Pet.java:112) ~[?:?]
        at be.isach.ultracosmetics.cosmetics.pets.Pet.onEquip(Pet.java:88) ~[?:?]
        at be.isach.ultracosmetics.cosmetics.Cosmetic.equip(Cosmetic.java:106) ~[?:?]
        at be.isach.ultracosmetics.cosmetics.type.CosmeticType.equip(CosmeticType.java:157) ~[?:?]
        at be.isach.ultracosmetics.cosmetics.type.PetType.equip(PetType.java:66) ~[?:?]
        at be.isach.ultracosmetics.cosmetics.type.PetType.equip(PetType.java:31) ~[?:?]
        at be.isach.ultracosmetics.menu.buttons.CosmeticButton.handleClick(CosmeticButton.java:119) ~[?:?]
        at be.isach.ultracosmetics.menu.buttons.CosmeticButton.onClick(CosmeticButton.java:79) ~[?:?]
        at be.isach.ultracosmetics.menu.Menu.onClick(Menu.java:182) ~[?:?]
        at jdk.internal.reflect.GeneratedMethodAccessor3.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:567) ~[?:?]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot-1.17.1.jar:3284a-Spigot-3892929-0ab8487]
        ... 18 more
```

### Additional Information

1.16.5 and 1.18.2 don't seem to be affected, seems like it's just a 1.17.1 issue

--- a/base/src/main/java/me/gamercoder215/mobchip/ai/behavior/CamelBehavior.java
+++ b/base/src/main/java/me/gamercoder215/mobchip/ai/behavior/CamelBehavior.java
@@ -1,0 +1,17 @@
+package me.gamercoder215.mobchip.ai.behavior;
+
+import me.gamercoder215.mobchip.ai.schedule.Updatable;
+
+/**
+ * Represents Entity Behaviors for a Camel
+ */
+public interface CamelBehavior extends CreatureBehavior, Updatable {
+
+    /**
+     * Makes this Camel sit.
+     * <p>This behavior does not require any memories.</p>
+     * @param minimalPoseTicks The minimum amount of ticks this camel should sit for
+     */
+    void sit(int minimalPoseTicks);
+
+}

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -14,7 +14,7 @@ plugins {
 }
 
 val pGroup = "me.gamercoder215"
-val pVersion = "1.9.2-SNAPSHOT"
+val pVersion = "1.9.3-SNAPSHOT"
 val pAuthor = "GamerCoder215"
 
 val github = "$pAuthor/MobChip"

--- a/bukkit/src/main/java/me/gamercoder215/mobchip/bukkit/BukkitBrain.java
+++ b/bukkit/src/main/java/me/gamercoder215/mobchip/bukkit/BukkitBrain.java
@@ -206,6 +206,12 @@ public class BukkitBrain implements EntityBrain {
 					constr = allay.getDeclaredConstructor(entityClass);
 					break;
 				}
+				case "camel": {
+					Class<? extends CamelBehavior> camel = Class.forName(BUKKIT_PACKAGE + "BukkitCamelBehavior")
+							.asSubclass(CamelBehavior.class);
+					constr = camel.getDeclaredConstructor(entityClass);
+					break;
+				}
 			}
 
 			constr.setAccessible(true);

--- a/nms/1_17_R1/src/main/java/me/gamercoder215/mobchip/abstraction/v1_17_R1/ChipUtil1_17_R1.java
+++ b/nms/1_17_R1/src/main/java/me/gamercoder215/mobchip/abstraction/v1_17_R1/ChipUtil1_17_R1.java
@@ -1167,7 +1167,10 @@ final class ChipUtil1_17_R1 implements ChipUtil {
 
     public static BlockPosition toNMS(Location l) { return new BlockPosition(l.getX(), l.getY(), l.getZ()); }
 
-    public static List<ItemStack> fromNMS(RecipeItemStack in) { return Arrays.stream(in.c).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList()); }
+    public static List<ItemStack> fromNMS(RecipeItemStack in) {
+        in.buildChoices();
+        return Arrays.stream(in.c).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList());
+    }
 
     public static Sound fromNMS(SoundEffect s) { return CraftSound.getBukkit(s); }
 

--- a/nms/1_20_R1/src/main/java/me/gamercoder215/mobchip/abstraction/v1_20_R1/ChipUtil1_20_R1.java
+++ b/nms/1_20_R1/src/main/java/me/gamercoder215/mobchip/abstraction/v1_20_R1/ChipUtil1_20_R1.java
@@ -61,6 +61,7 @@ import net.minecraft.world.entity.ambient.AmbientCreature;
 import net.minecraft.world.entity.animal.*;
 import net.minecraft.world.entity.animal.allay.AllayAi;
 import net.minecraft.world.entity.animal.axolotl.AxolotlAi;
+import net.minecraft.world.entity.animal.camel.CamelAi;
 import net.minecraft.world.entity.animal.frog.FrogAi;
 import net.minecraft.world.entity.boss.enderdragon.EndCrystal;
 import net.minecraft.world.entity.boss.enderdragon.phases.*;
@@ -1628,6 +1629,7 @@ final class ChipUtil1_20_R1 implements ChipUtil {
         else if (c instanceof Axolotl) AxolotlAi.updateActivity((net.minecraft.world.entity.animal.axolotl.Axolotl) nms);
         else if (c instanceof Frog) FrogAi.updateActivity((net.minecraft.world.entity.animal.frog.Frog) nms);
         else if (c instanceof org.bukkit.entity.Warden) WardenAi.updateActivity((Warden) nms);
+        else if (c instanceof Camel) CamelAi.updateActivity((net.minecraft.world.entity.animal.camel.Camel) nms);
     }
 
     public static MemoryModuleType<?> toNMS(Memory<?> mem) {

--- a/nms/1_20_R1/src/main/java/me/gamercoder215/mobchip/bukkit/BukkitCamelBehavior.java
+++ b/nms/1_20_R1/src/main/java/me/gamercoder215/mobchip/bukkit/BukkitCamelBehavior.java
@@ -1,0 +1,19 @@
+package me.gamercoder215.mobchip.bukkit;
+
+import me.gamercoder215.mobchip.ai.behavior.CamelBehavior;
+import org.bukkit.entity.Camel;
+
+class BukkitCamelBehavior extends BukkitUpdatableCreatureBehavior implements CamelBehavior {
+
+    final Camel m;
+
+    public BukkitCamelBehavior(Camel m) {
+        super(m);
+        this.m = m;
+    }
+
+    @Override
+    public void sit(int minimalPoseTicks) {
+        wrapper.runBehavior(m, "CamelAi$RandomSitting", "net.minecraft.world.entity.animal.camel", minimalPoseTicks);
+    }
+}

--- a/nms/1_20_R2/src/main/java/me/gamercoder215/mobchip/abstraction/v1_20_R2/ChipUtil1_20_R2.java
+++ b/nms/1_20_R2/src/main/java/me/gamercoder215/mobchip/abstraction/v1_20_R2/ChipUtil1_20_R2.java
@@ -61,6 +61,7 @@ import net.minecraft.world.entity.ambient.AmbientCreature;
 import net.minecraft.world.entity.animal.*;
 import net.minecraft.world.entity.animal.allay.AllayAi;
 import net.minecraft.world.entity.animal.axolotl.AxolotlAi;
+import net.minecraft.world.entity.animal.camel.CamelAi;
 import net.minecraft.world.entity.animal.frog.FrogAi;
 import net.minecraft.world.entity.boss.enderdragon.EndCrystal;
 import net.minecraft.world.entity.boss.enderdragon.phases.*;
@@ -1628,6 +1629,7 @@ final class ChipUtil1_20_R2 implements ChipUtil {
         else if (c instanceof Axolotl) AxolotlAi.updateActivity((net.minecraft.world.entity.animal.axolotl.Axolotl) nms);
         else if (c instanceof Frog) FrogAi.updateActivity((net.minecraft.world.entity.animal.frog.Frog) nms);
         else if (c instanceof org.bukkit.entity.Warden) WardenAi.updateActivity((Warden) nms);
+        else if (c instanceof Camel) CamelAi.updateActivity((net.minecraft.world.entity.animal.camel.Camel) nms);
     }
 
     public static MemoryModuleType<?> toNMS(Memory<?> mem) {

